File: Java/BookstoreAPI/src/main/java/com/bookstore/endpoints/GenreService.java
package com.bookstore.endpoints;

import com.bookstore.BookstoreRESTApp;
import com.bookstore.models.Genre;
import io.swagger.annotations.Api;
import org.springframework.stereotype.Component;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestParam;

import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;
import java.util.List;

/**
 * Created by GOA on 7/10/16.
 */
@Component
@Path("/api/genres")
@Api(value = "GenreService", produces = "application/json")
public class GenreService {
    @GET
    @Path("")
    @Produces(MediaType.APPLICATION_JSON)
    public List<Genre> getAllGenres() {
        return BookstoreRESTApp.getInstance().getGenreRepository().findAll();
    }

    @POST
    @Path("/")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    public String addGenre(@RequestBody Genre genre) {
        BookstoreRESTApp.getInstance().getGenreRepository().insert(genre);
        return "OK";
    }

    @PUT
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    @Consumes(MediaType.APPLICATION_JSON)
    public String updateGenre(@RequestBody Genre genre) {
        BookstoreRESTApp.getInstance().getGenreRepository().save(genre);
        return "OK";
    }

    @DELETE
    @Path("{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public String deleteGenre(@PathParam("id") String id) {
        BookstoreRESTApp.getInstance().getGenreRepository().delete(BookstoreRESTApp.getInstance().getGenreRepository().findOne(id));
        return "OK";
    }
}

File: Node.js/bookstore/public/js/services/GenreService.js
var genreService = angular.module("genreService", []);

genreService.factory("GenreService", function($http) {
  var services = {};

  services.getAllGenres = function() {
    return $http.get('http://localhost:8081/api/genres').then(function(response) {
        return response.data;
    });
  };

  services.deleteGenreById = function(id) {
    return $http.delete('http://localhost:8081/api/genres/'+id).then(function(response) {
        return response.data;
    });
  }

  services.addGenre = function(genre) {
    return $http.post('http://localhost:8081/api/genres', genre).then(function(response) {
        return response.data;
    })
  }

  services.updateGenre = function(genre) {
    return $http.put('http://localhost:8081/api/genres/' + genre._id, genre).then(function(response) {
        return response.data;
    })
  }

  return services;
});
